react/features/base/dialog/components/BottomSheet.native.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import React, { Component, type Node } from 'react';
+import { Modal, TouchableWithoutFeedback, View } from 'react-native';
+
+import { bottomSheetStyles as styles } from './styles';
+
+/**
+ * The type of {@code BottomSheet}'s React {@code Component} prop types.
+ */
+type Props = {
+
+    /**
+     * The children to be displayed within this component.
+     */
+    children: Node,
+
+    /**
+     * Handler for the cancel event, which happens when the user dismisses
+     * the sheet.
+     */
+    onCancel: Function
+};
+
+/**
+ * A component emulating Android's BottomSheet. For all intents and purposes,
+ * this component has been designed to work and behave as a {@code Dialog}.
+ */
+export default class BottomSheet extends Component<Props> {
+    /**
+     * Initializes a new {@code BottomSheet} instance.
+     *
+     * @param {Object} props - The read-only React {@code Component} props with
+     * which the new instance is to be initialized.
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._onCancel = this._onCancel.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return [
+            <View
```

Comment:
Right. I tried to explain it as best as I could, but I guess I failed :-)

We want to have a backdrop, which is clickable, and when clicked the modal would be dismissed. The slide effect is part of the modal, and if we make the backdrop part of it, it looks weird because it slides with it.

By using a separate view, the overlay won't move, and the backdrop is just a transparent view which sees through to the overlay.

Makes sense?

Reply:
Absolutely, thanks for explaining!